src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -0,0 +1,932 @@
+    // the number to > 1, otherwise importing the music collection
+    // takes place when the GUI threads terminates, i.e., on
+    // Mixxx shutdown.
+    QThreadPool::globalInstance()->setMaxThreadCount(4); //Tobias decided to use 4
+    // Let a worker thread do the XML parsing
+    m_devicesFuture = QtConcurrent::run(this, &RekordboxFeature::findRekordboxDevices);
+    m_devicesFutureWatcher.setFuture(m_devicesFuture);
+    m_title = tr("(loading) Rekordbox");
+    //calls a slot in the sidebar model such that 'Rekordbox (isLoading)' is displayed.
+    emit(featureIsLoading(this, true));
+
+    emit(enableCoverArtDisplay(true));  
+    emit(switchToView("REKORDBOXHOME"));
+   
+}
+
+void RekordboxFeature::activateChild(const QModelIndex& index) {
+    if (!index.isValid()) return;
+
+    //access underlying TreeItem object
+    TreeItem *item = static_cast<TreeItem*>(index.internalPointer());
+    if (!(item && item->getData().isValid())) {
+        return;
+    }
+
+    // TreeItem list data holds 2 values in a QList and have different meanings.
+    // If the 2nd QList element IS_RECORDBOX_DEVICE, the 1st element is the 
+    // filesystem device path, and the parseDeviceDB concurrent thread to parse
+    // the Rekcordbox database is initiated. If the 2nd element is 
+    // IS_NOT_RECORDBOX_DEVICE, the 1st element is the playlist path and it is
+    // activated.
+    QList<QVariant> data = item->getData().toList();
+    QString playlist = data[0].toString();
+    bool doParseDeviceDB = data[1].toString() == IS_RECORDBOX_DEVICE;
+
+    qDebug() << "RekordboxFeature::activateChild " << item->getLabel()
+             << " playlist: " << playlist << " doParseDeviceDB: " << doParseDeviceDB;
+
+    if (doParseDeviceDB) {
+        qDebug() << "Parse Rekordbox Device DB: " << playlist;
+
+        // Usually the maximum number of threads
+        // is > 2 depending on the CPU cores
+        // Unfortunately, within VirtualBox
+        // the maximum number of allowed threads
+        // is 1 at all times We'll need to increase
+        // the number to > 1, otherwise importing the music collection
+        // takes place when the GUI threads terminates, i.e., on
+        // Mixxx shutdown.
+        QThreadPool::globalInstance()->setMaxThreadCount(4); //Tobias decided to use 4
+        // Let a worker thread do the XML parsing
+        m_tracksFuture = QtConcurrent::run(this, &RekordboxFeature::parseDeviceDB, item);
+        m_tracksFutureWatcher.setFuture(m_tracksFuture);
+
+        // This device is now a playlist element, future activations should treat is
+        // as such
+        data[1] = QVariant(IS_NOT_RECORDBOX_DEVICE);
+        item->setData(QVariant(data));
+    } else {
+        qDebug() << "Activate Rekordbox Playlist: " << playlist;
+        m_pRekordboxPlaylistModel->setPlaylist(playlist);
+        emit(showTrackModel(m_pRekordboxPlaylistModel));
+    }
+}
+
+QList<TreeItem*> RekordboxFeature::findRekordboxDevices() {
```

Comment:
Does this function need to be a member function? Instead use free functions defined in an anonymous namespace if possible. Also check all other member functions if they really dependon the state of the object instance.